pkg/graph/resolvers/key_contact.go

```diff
@@ -0,0 +1,320 @@
+			logger,
+			mailboxContact,
+		)
+
+		if err != nil {
+			return nil, err
+		}
+
+		return newContact, nil
+	})
+
+	if err != nil {
+		return nil, err
+	}
+
+	if emailService != nil && emailTemplateService != nil {
+		go func() {
+			sendEmailErr := sendKeyContactWelcomeEmail(
+				emailService,
+				emailTemplateService,
+				addressBook,
+				newContact,
+			)
+			if sendEmailErr != nil {
+				logger.Error(
+					"failed to send key contact welcome email for create by mailbox",
+					zap.String("mailboxAddress", mailboxAddress),
+					zap.Error(sendEmailErr),
+				)
+			}
+		}()
+	}
+
+	return newContact, nil
+}
+
+// UpdateKeyContact updates an existing user or mailbox contact for a subject matter expert.
+// Only subjectCategoryID, subjectArea, and mailboxTitle fields can be changed. Returns the updated contact.
+func UpdateKeyContact(ctx context.Context, logger *zap.Logger, principal authentication.Principal, store *storage.Store,
+	emailService oddmail.EmailService, emailTemplateService email.TemplateService, addressBook email.AddressBook,
+	id uuid.UUID,
+	changes map[string]interface{},
+) (*models.KeyContact, error) {
+	principalAccount := principal.Account()
+	if principalAccount == nil {
+		return nil, fmt.Errorf("principal doesn't have an account, username %s", principal.String())
+	}
+
+	existingContact, err := loaders.KeyContact.ByID.Load(ctx, id)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get contact with id %s: %w", id, err)
+	}
+	if existingContact == nil {
+		return nil, fmt.Errorf("contact with id %s not found", id)
+	}
+
+	err = BaseStructPreUpdate(logger, existingContact, changes, principal, store, true, false)
+	if err != nil {
+		return nil, err
+	}
+
+	updatedContact, err := sqlutils.WithTransaction(store, func(tx *sqlx.Tx) (*models.KeyContact, error) {
+		updatedContact, err := storage.KeyContactUpdateContact(tx, logger, existingContact)
+		if err != nil {
+			return nil, fmt.Errorf("failed to update contact with id %s: %w", id, err)
+		}
+
+		return updatedContact, nil
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	return updatedContact, nil
+}
+
+// DeleteKeyContact deletes a subject matter expert by its ID.
+// Returns the deleted contact or an error.
+func DeleteKeyContact(ctx context.Context, logger *zap.Logger, principal authentication.Principal, store *storage.Store,
+	emailService oddmail.EmailService, emailTemplateService email.TemplateService, addressBook email.AddressBook,
+	id uuid.UUID,
+) (*models.KeyContact, error) {
+	principalAccount := principal.Account()
+	if principalAccount == nil {
+		return nil, fmt.Errorf("principal doesn't have an account, username %s", principal.String())
+	}
+
+	// Use a transaction for delete (for audit triggers, etc.)
+	returnedContact, err := sqlutils.WithTransaction(store, func(tx *sqlx.Tx) (*models.KeyContact, error) {
+		// Fetch the existing contact to check permissions and return after delete
+		existing, err := loaders.KeyContact.ByID.Load(ctx, id)
```

Comment:
nitpick: the loader is transactionally unaware, since it's not built to be used to pass in transaction (tx) to, I would move it out of the scope of the transaction to make that difference more evident